lib/utils/nodeFieldIndices.mjs

```diff
@@ -0,0 +1,118 @@
+import getAtRuleParams from './getAtRuleParams.mjs';
+import getRuleSelector from './getRuleSelector.mjs';
+
+import { isObject, isString } from './validateTypes.mjs';
+
+/** @import {AtRule, Declaration, Rule} from 'postcss' */
+
+/**
+ * @param {AtRule} atRule
+ * @returns {number}
+ */
+export function atRuleParamIndex(atRule) {
+	let index = atRuleRawsAfterNameIndex(atRule);
+
+	if (atRule.raws.afterName) {
+		index += atRule.raws.afterName.length;
+	}
+
+	return index;
+}
+
+/**
+ * @param {AtRule} node
+ * @returns {number}
+ */
+export function atRuleRawsAfterIndex(node) {
+	// subtract 1 for `}`
+
+	if (!node.source?.end?.offset) return node.toString().length - 1;
+
+	if (!node.raws?.after?.length) return node.source.end.offset - 1;
+
+	return node.source.end.offset - (node.raws.after.length + 1);
+}
+
+/**
+ * @param {AtRule} atRule
+ * @returns {number}
+ */
+export function atRuleRawsAfterNameIndex(atRule) {
+	// Initial 1 is for the `@`
+	return 1 + atRule.name.length;
+}
+
+/**
+ * @param {AtRule} atRule
+ * @returns {number}
+ */
+export function atRuleRawsBetweenIndex(atRule) {
+	return atRuleParamIndex(atRule) + getAtRuleParams(atRule).length;
+}
+
+/**
+ * @param {Declaration} decl
+ * @returns {number}
+ */
+export function declarationRawsBetweenIndex(decl) {
+	const raws = decl.raws;
+	const prop = raws.prop;
+
+	return [
+		isObject(prop) && 'prefix' in prop && prop.prefix,
+		(isObject(prop) && 'raw' in prop && prop.raw) || decl.prop,
+		isObject(prop) && 'suffix' in prop && prop.suffix,
+	].reduce((/** @type {number} */ count, str) => {
+		if (isString(str)) {
+			return count + str.length;
+		}
+
+		return count;
+	}, 0);
```

Comment:
[suggestion] How about extracting a function from this `reduce()` part for readability? For example:

```js
/**
 * @param {unknown[]} values
 * @returns {number}
 */
function countChars(values) {
	return values.reduce((/** @type {number} */ count, value) => {
		if (isString(value)) return count + value.length;

		return count;
	}, 0);
}
```

Would change to:

```js
	return countChars([
		propIsObject && 'prefix' in prop && prop.prefix,
		(propIsObject && 'raw' in prop && prop.raw) || decl.prop,
		propIsObject && 'suffix' in prop && prop.suffix,
	]);
```